### PR TITLE
feat(config): better config migration

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Improve preset migration/replacement.

## Context:

We need the ability to both migrate as well as remove presets.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/index.ts [hunk 1]
```diff
@@ -154,11 +154,43 @@ export function parsePreset(input: string): ParsedPreset {
...
   baseConfig?: RenovateConfig
 ): Promise<RenovateConfig> {
   logger.trace(`getPreset(${preset})`);
+  // Check if the preset has been removed or replaced
+  const newPreset = removedPresets[preset];
+  if (newPreset) {
+    return getPreset(newPreset, baseConfig);
+  }
+  if (newPreset === null) {
+    return {};
```
Comment (viceice) [inline lib/config/presets/index.ts:192]: I think we should warn if an old preset is used.

**Review by viceice — COMMENTED** [review]
Needs test and lint fixes

File: lib/config/presets/index.ts (hunk 1 above)
Comment (viceice) [reply lib/config/presets/index.ts:169]: ⬆️ ❓

Comment (rarkins) [reply lib/config/presets/index.ts:169]: I didn't do that as we will migrate such unused presets out in future with a PR anyway. I don't think we need to warn people if e.g. we have a preset to enable an experimental feature and later it becomes default.

**Review by viceice — APPROVED** [review]

**Pull request merged**